### PR TITLE
✨ Add `ids` option for `needimport`

### PR DESCRIPTION
**Pull request by chrisjsewell**
A more performant alternative to the `filter` option.

This PR also adds some other performance optimisations;
caching the schema load and not reading the import file twice (during validation)

Also, add to and improve the `needimport` tests

### CODE REVIEW

File: sphinx_needs/directives/needimport.py [hunk 1]
```diff
@@ -115,21 +112,21 @@ def run(self) -> Sequence[nodes.Node]:
                     f"Could not load needs import file {correct_need_import_path}"
                 )
 
-            errors = check_needs_file(correct_need_import_path)
+            try:
+                with open(correct_need_import_path) as needs_file:
+                    needs_import_list = json.load(needs_file)
+            except json.JSONDecodeError as e:
```
Comment (ubmarco) [inline sphinx_needs/directives/needimport.py:118]: `OSError` should also be caught, but there is already a TODO

File: sphinx_needs/directives/needimport.py [hunk 2]
```diff
@@ -146,6 +143,13 @@ def run(self) -> Sequence[nodes.Node]:
 
         needs_config = NeedsSphinxConfig(self.config)
         data = needs_import_list["versions"][version]
+
+        if ids := self.options.get("ids"):
+            id_list = [i.strip() for i in ids.split(",") if i.strip()]
```
Comment (ubmarco) [inline sphinx_needs/directives/needimport.py:148]: Can IDs contain a dynamic function, so you would have to use the 'split with dyn' function?

File: sphinx_needs/directives/needimport.py [hunk 3]
```diff
@@ -184,8 +188,13 @@ def run(self) -> Sequence[nodes.Node]:
         needs_list = needs_list_filtered
 
         # tags update
-        for need in needs_list.values():
-            need["tags"] = need["tags"] + tags
+        if tags := [
+            tag.strip()
+            for tag in re.split("[;,]", self.options.get("tags", ""))
```
Comment (ubmarco) [inline sphinx_needs/directives/needimport.py:193]: same here, the new function should be used (or is that in a different open PR?)

**Review by ubmarco — CHANGES_REQUESTED** [review]

File: sphinx_needs/directives/needimport.py (hunk 2 above)
Comment (chrisjsewell) [reply sphinx_needs/directives/needimport.py:148]: no definitely not, no dynamic functions here; this is intended for fast filtering

File: sphinx_needs/directives/needimport.py (hunk 3 above)
Comment (chrisjsewell) [reply sphinx_needs/directives/needimport.py:193]: this was already existing, but again no dynamic functions.
I feel you are confusing this directive with a `need` declaration directive, that is the only place we use dynamic functions in the directive options

File: sphinx_needs/directives/needimport.py (hunk 1 above)
Comment (chrisjsewell) [inline sphinx_needs/directives/needimport.py:118]:
```suggestion
            except (OSError, json.JSONDecodeError) as e:
```

Comment (chrisjsewell) [reply sphinx_needs/directives/needimport.py]: done

Comment (chrisjsewell) [reply sphinx_needs/directives/needimport.py]: (knowing what exceptions to handle is always a pain, I was reading this with interest recently https://discuss.python.org/t/extend-type-hints-to-cover-exceptions/23788/51)

**Review by ubmarco — APPROVED** [review]
Thanks for the PR

**Pull request merged**